lib/rspec/core.rb

```diff
@@ -31,6 +31,7 @@
   option_parser
   configuration_options
   runner
+  profiler
```

Comment:
>A big reason tracking example group execution time in a registered formatter (even if you don't call it one but I do!) 

In our terms a formatter is responsible for output to the user, in some cases they format the data themselves in others they use the presenter pattern to share workload between them, the Profiler does neither of these, it neither outputs anything or formats anything, it **just** captures values. It's no more a formatter than the reporter, or the runner or any other core component of RSpec.

The reporter follows (roughly) the event bus pattern, it sends out events when triggered to any registered **listener** of which the primary example are formatters I will admit, but this is not a formatter. In the same way your exception presenter wasn't a formatter. Naming is important.

> appealed to me is that we want the added runtime cost of tracking times for profiling to only be paid if the user has opted in via --profile. It makes me a sad panda to always pay the cost of tracking that when the user hasn't turned on profiling.

> IMO, we don't need to support dynamic changes to config#profile_examples after the suite has started running. We can treat it like adding a formatter: it has to be done before the suite starts to take affect. (A user obviously can't expect to turn on profiling half way through the suite run and expect it to work).

Thats fine but theres probably edge cases involved and I'd rather go back to setting it at the start, off or on, no half way ground (because otherwise the profiler has to go back to config, or we have to setup two way communication between the config and the reporter (to turn on and off the profiler), which I'd rather not do.

> My problem with what you had before is that config exposed the profile_examples option and a derived attribute, and it seems bad for an object to expose a derived attribute but then not keep it in sync with the source value. Now that you're not exposing the profiler object off of config, I'm not concerned about that anymore. Now I just view profile_examples as a boolean toggle that is checked once, when setting things up to start running the suite, and ignored the rest of the time. Users can toggle it back and forth all they want before the suite starts and it'll work fine still.

They won't be able to do that if the Profiler / NoProfiler is initialised in the reporter unless we somehow hook into the repoter from the config, I'd rather we set it once.

> In effect, I think we should only load the profiler and register it with the reporter if we get here. That'll avoid the cost of loading and using it if the user hasn't opted-in to it.